### ISSUE TITLE
Context menus block the renderer

### ISSUE THREAD

**Issue by obrejla** [Microsoft/vscode#23085]
Hi,

- VSCode Version: 1.10.2
- OS Version: macOS

Steps to Reproduce:

1. run r.g. test in terminal which prints a lot of output
2. during the run right click some open file tab to see its context menu
3. profit...printing output to terminal stops.

And what's worse, when you hide that context menu, the printing continues again, so far so good, but the output which should be printed during the context menu appearance is **not** printed to terminal output. Only new output continues to be printed.

Thanks.

**Comment by Njanderson** [comment]
I'll take a look at this.

**Comment by Tyriar** [comment]
I created a minimal repro case in Electron and created https://github.com/electron/electron/issues/11512

@bpasero FYI in case there are any duplicates of this

**Comment by bpasero** [comment]
@Tyriar this is not an issue with Electron, this is an issue with us not using the `async` option for context menus. We cannot easily use `async` because we rely on a way to find out if the menu was closed or not and this is currently not provided by electron (see https://github.com/electron/electron/issues/9441)

**Comment by Tyriar** [comment]
@bpasero shouldn't we keep the electron/upstream labels as it's blocked by https://github.com/electron/electron/issues/9441?

**Comment by bpasero** [comment]
@Tyriar yeah why not

**Comment by bpasero** [comment]
Closing as the update to Electron 2.0.0-beta.6 landed via https://github.com/Microsoft/vscode/commit/88603b377d56ead7b5c4cfc6bc3a64e5f832cd60. It should be in insider releases end of this week.

**Comment by bpasero** [comment]
Reopening as we had to revert the update.

**Comment by bpasero** [comment]
You can verify by seeing the cursor blink when you right click (macOS with native menus).
